caravel/assets/javascripts/explorev2/reducers.js

```diff
@@ -0,0 +1,127 @@
+import shortid from 'shortid';
+import * as actions from './actions';
+
+const defaultTimeFilter = {
+  timeColumn: null,
+  timeGrain: null,
+  since: null,
+  until: null,
+};
+
+const defaultGroupBy = {
+  groupByColumn: [],
+  metrics: [],
+};
+
+const defaultSql = {
+  where: '',
+  having: '',
+};
+
+const defaultFilter = {
+  id: shortid.generate(),
+  eq: null,
+  op: 'in',
+  col: null,
+};
+
+export const initialState = {
+  datasourceId: null,
+  vizType: null,
+  timeFilter: defaultTimeFilter,
+  groupBy: defaultGroupBy,
+  columns: [],
+  orderings: [],
+  timeStampFormat: null,
+  rowLimit: null,
+  searchBox: false,
+  SQL: defaultSql,
+  filters: [defaultFilter],
+};
+
+function addToArr(state, arrKey, obj) {
```

Comment:
yes, definitely great idea to share these utils. best place for them would be `caravel/assets/javascripts/utlils/`, then all reducers can import from there. 

reducerUtils.js
```
export function addToArr(state, arrKey, obj) {
  ...
}
```

`import { addToArr } from '../utils/reducerUtils.js'